src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -396,6 +395,9 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 			if (index !== -1) {
 				this.tabToolbarActionsConfig.set(value.id, configs[index].widget['tasks-widget']);
 				const content = this.getToolbarContent(configs[index].widget['tasks-widget']);
+				if (content.length > 0) {
```

Comment:
Why only if there's other actions? I thought you were going to have refresh always show?

Reply:
Discussed this with Alan and changed this so that refresh will show in the toolbar if the tab content is a widgets container, grid container, or nav section because those have refresh implemented and will refresh the widgets. If an extension is using something else for the content, like modelview container, they should define their own refresh.